src/app/components/OrderInput.tsx

```diff
@@ -451,13 +454,60 @@ function SubmitButton() {
 }
 
 function UserInputContainer() {
-  const { side, type } = useAppSelector((state) => state.orderInput);
+  const dispatch = useAppDispatch();
+  const { side, type, token1, token2 } = useAppSelector(
+    (state) => state.orderInput
+  );
+  const balanceToken1 =
+    useAppSelector((state) => selectBalanceByAddress(state, token1.address)) ||
+    0;
+  const balanceToken2 =
+    useAppSelector((state) => selectBalanceByAddress(state, token2.address)) ||
+    0;
+  const bestBuy = useAppSelector((state) => state.orderBook.bestBuy) || 0;
+  const bestSell = useAppSelector((state) => state.orderBook.bestSell) || 0;
 
   const isMarketOrder = type === "MARKET";
   const isLimitOrder = type === "LIMIT";
   const isBuyOrder = side === "BUY";
   const isSellOrder = side === "SELL";
 
+  useEffect(() => {
+    if (isMarketOrder) {
+      sliderCallback(0, "MARKET");
+    } else if (isLimitOrder) {
+      sliderCallback(0, "LIMIT");
+    }
+  }, [isBuyOrder, isSellOrder, isMarketOrder, isLimitOrder]);
+
+  const sliderCallback = (newPercentage: number, type: string) => {
+    console.log(`Type is: ${type}`);
+    const isXRDToken = isBuyOrder
+      ? token2.symbol === "XRD"
+      : token1.symbol === "XRD";
+    let balance = isBuyOrder ? balanceToken2 : balanceToken1;
+
+    if (newPercentage === 100 && isXRDToken) {
+      balance = Math.max(balance - XRD_FEE_ALLOWANCE, 0);
+    }
+
+    const amount = (balance * newPercentage) / 100;
```

Comment:
Please use our Calculator class for any calculations. The reason is that javascript cannot precisely calculate (you can see that by typing in 0.1+0.2 in a console, the result is surprising. 

Any operation (addition, substraction, division, multiplication) should be done using our [`Calculator`](https://github.com/DeXter-on-Radix/website/blob/main/src/app/services/Calculator.ts) class.

```suggestion
    const amount = Calculator.divide(Calculator.multiply(balance, newPercentage), 100);
```